client/state/notices/actionCreators.js

```diff
@@ -0,0 +1,27 @@
+import { createNoticeAction, removeNoticeAction } from './actions';
+
+export default function( dispatch ) {
+	function createNotice( type, text, options ) {
+		var action = createNoticeAction( type, text, options );
+
+		if ( action.duration > 0 ) {
+			setTimeout( () => {
```

Comment:
@gziolo said: in #1041 :
> Another param that can cause troubles is duration. Though in this case we can trigger action using component lifecycle methods. I'm not sure if that's good practice, but we could use componentDidMount and set timeout which would trigger action after given duration.

I know this approach here does not survive dehydration.
We will need to approach it somehow, but I dont want to get component involved at this point.